patches/post/20231023-Redact-auth-token-in-debug-output.patch

```diff
@@ -0,0 +1,67 @@
+diff --git a/metal/v1/client.go b/metal/v1/client.go
+index 082bc38..46b7665 100644
+--- a/metal/v1/client.go
++++ b/metal/v1/client.go
+@@ -355,6 +355,9 @@ func parameterToJson(obj interface{}) (string, error) {
+ 
+ // callAPI do the request.
+ func (c *APIClient) callAPI(request *http.Request) (*http.Response, error) {
++    if len(request.Header.Get("X-Auth-Token")) != 0 {
++        request.Header.Set("X-Auth-Token", "**REDACTED**")
```

Comment:
This reads like it will overwrite the token before sending the request to the API, in which case every request is guaranteed to fail.

Reply:
Using a custom template would involve

- adding this template to the `templates` directory: https://github.com/OpenAPITools/openapi-generator/blob/v7.0.0/modules/openapi-generator/src/main/resources/go/client.mustache
- modifying the template to do the things that we did in the patch
- deleting the patch

Re: overwriting the request header, note that packngo cloned the request and then modified the clone in order to redact the token in output without impacting the actual request: https://github.com/packethost/packngo/blob/d7f968e2791e84e64be19616765bee2503fef42a/packngo.go#L293